pkg/promclient/promclient.go

```diff
@@ -88,6 +89,157 @@ func ExternalLabels(ctx context.Context, logger log.Logger, base *url.URL) (labe
 	return labels.FromMap(cfg.Global.ExternalLabels), nil
 }
 
+type Flags struct {
+	TSDBPath           string         `json:"storage.tsdb.path"`
+	TSDBRetention      model.Duration `json:"storage.tsdb.retention"`
+	TSDBMinTime        model.Duration `json:"storage.tsdb.min-block-duration"`
+	TSDBMaxTime        model.Duration `json:"storage.tsdb.max-block-duration"`
+	WebEnableAdminAPI  bool           `json:"web.enable-admin-api"`
+	WebEnableLifecycle bool           `json:"web.enable-lifecycle"`
+}
+
+// UnmarshalJSON implements the json.Unmarshaler interface.
+func (f *Flags) UnmarshalJSON(b []byte) error {
+	// TODO(bwplotka): Avoid this custom unmarshal by:
+	// - prometheus/common: adding unmarshalJSON to modelDuration
+	// - prometheus/prometheus: flags should return proper JSON. (not bool in string)
+	parsableFlags := struct {
+		TSDBPath           string        `json:"storage.tsdb.path"`
+		TSDBRetention      modelDuration `json:"storage.tsdb.retention"`
+		TSDBMinTime        modelDuration `json:"storage.tsdb.min-block-duration"`
+		TSDBMaxTime        modelDuration `json:"storage.tsdb.max-block-duration"`
+		WebEnableAdminAPI  modelBool     `json:"web.enable-admin-api"`
+		WebEnableLifecycle modelBool     `json:"web.enable-lifecycle"`
+	}{}
+
+	if err := json.Unmarshal(b, &parsableFlags); err != nil {
+		return err
+	}
+
+	*f = Flags{
+		TSDBPath:           parsableFlags.TSDBPath,
+		TSDBRetention:      model.Duration(parsableFlags.TSDBRetention),
+		TSDBMinTime:        model.Duration(parsableFlags.TSDBMinTime),
+		TSDBMaxTime:        model.Duration(parsableFlags.TSDBMaxTime),
+		WebEnableAdminAPI:  bool(parsableFlags.WebEnableAdminAPI),
+		WebEnableLifecycle: bool(parsableFlags.WebEnableLifecycle),
+	}
+	return nil
+}
+
+type modelDuration model.Duration
+
+// UnmarshalJSON implements the json.Unmarshaler interface.
+func (d *modelDuration) UnmarshalJSON(b []byte) error {
+	var s string
+	if err := json.Unmarshal(b, &s); err != nil {
+		return err
+	}
+
+	dur, err := model.ParseDuration(s)
+	if err != nil {
+		return err
+	}
+	*d = modelDuration(dur)
+	return nil
+}
+
+type modelBool bool
+
+// UnmarshalJSON implements the json.Unmarshaler interface.
+func (m *modelBool) UnmarshalJSON(b []byte) error {
+	var s string
+	if err := json.Unmarshal(b, &s); err != nil {
+		return err
+	}
+
+	boolean, err := strconv.ParseBool(s)
+	if err != nil {
+		return err
+	}
+	*m = modelBool(boolean)
+	return nil
+}
+
+// ConfiguredFlags some configured flags from /api/v1/status/flags Prometheus endpoint.
+// Added to Prometheus from v2.2.
+func ConfiguredFlags(ctx context.Context, logger log.Logger, base *url.URL) (Flags, error) {
```

Comment:
Is this only used in the e2e tests?

Reply:
For now yes, but will used in more places